Reader: Follow button fixes

wargcm opened the pull request:
Fixes #21809

## Description

- Fixes the font not being set on the follow buttons
- Reduces the content inset of the follow button on the followed sites list

## Testing

To test:
- Launch Jetpack and login
- Tap on Reader
- Tap on `Discover`
- Scroll down until you see a `Sites to follow` cell
- 🔎 **Verify** the `Follow` text is using the 15pt font size
- Tap on the manage icon in the top right
- Tap on `Followed Sites`
- 🔎 **Verify** the leading and trailing content insets are reduced to 16pt each

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

wargcm committed:
Fix Reader's follow button font size

wargcm committed:
Reduce insets on follow button in sites list

wargcm commented on WordPress/Classes/Extensions/Font/UIConfigurationTextAttributesTransformer+Font.swift:
```diff
@@ -0,0 +1,13 @@
+extension UIConfigurationTextAttributesTransformer {
+
+    /// Instance method that sets the font on the ``AttributeContainer``
+    static func transformer(with font: UIFont) -> UIConfigurationTextAttributesTransformer {
+        return UIConfigurationTextAttributesTransformer { incoming in
+            var outgoing = incoming
+            outgoing.font = font
+            return outgoing
+        }
+    }
```
There are a few other places that use the same code, which is why I made this extension. I didn't update those locations though since this is going to a release branch and don't want to introduce more risk than necessary.

dvdchr replied:
>  I didn't update those locations though since this is going to a release branch

Good point. 👍🏼

dvdchr reviewed (commented):
Changes are looking good, but the "Sites to follow" should also use the 16pt horizontal insets. The only places where the follow button has 24pt insets are the Tag and Sites stream header, based on p1697541472738849-slack-C05N140C8H5.

wargcm committed:
Update follow content inset on `Sites to follow`

wargcm commented:
@dvdchr Ah, you're right. I misunderstood and thought "lists" was referring to the followed sites list, not all lists. I fixed it in https://github.com/wordpress-mobile/WordPress-iOS/pull/21810/commits/cd0e0df6d2cdf9e095d8df2883d778af6c9fd5e8. I also updated the new detail header in https://github.com/wordpress-mobile/WordPress-iOS/pull/21810/commits/5b02cd21b844b1402e73e186b5801ed12dd43ac4. Thanks for the catch!

wargcm committed:
Reduce follow padding for new detail header

dvdchr reviewed (approved):
LGTM! :shipit:
